libvast/src/system/spawn_node.cpp

```diff
@@ -52,11 +52,14 @@ spawn_node(caf::scoped_actor& self, const caf::settings& opts) {
   // output is written into the same directory.
   if (auto err = initialize_db_version(abs_dir))
     return err;
-  // TODO(ch20326): Replace this with a more specific check in the components
-  // that rely on a specific DB version.
-  if (read_db_version(abs_dir) != db_version::latest)
+  if (auto version = read_db_version(abs_dir); version != db_version::latest) {
+    VAST_INFO_ANON("Cannot start VAST, breaking changes detected in the "
+                   "database directory");
+    auto reasons = describe_breaking_changes_since(version);
+    VAST_INFO_ANON("Detailed explanation:\n", reasons);
```

Comment:
Output-wise, wouldn't it make more sense to format the output as a list of breaking changes starting with dashes?

Also I think downstream should print this error message such that we're not relying on the log level info to be available. The following should work (roughly):

```
    auto reasons = describe_breaking_changes_since(version);
    return make_error(ec::breaking_change, "breaking changes detected since last start of the VAST with the current database directory:", reasons);
```